jcl/src/java.base/share/classes/java/lang/Thread.java

```diff
@@ -1172,19 +1177,18 @@ public synchronized void start() {
 	 *
 	 * Release the lock before calling threadgroup's remove method for this thread.
 	/*[ENDIF]*/
+	synchronized (lock) {
+		if (started) {
+			/*[MSG "K0341", "Thread is already started"]*/
+			throw new IllegalThreadStateException(com.ibm.oti.util.Msg.getString("K0341")); //$NON-NLS-1$
+		}
+		/*[PR 115667, CMVC 94448] In 1.5, thread is added to ThreadGroup when started */
+		group.add(this);
+	}
 
 	try {
-		synchronized(lock) {
-			if (started) {
-				/*[MSG "K0341", "Thread is already started"]*/
-				throw new IllegalThreadStateException(com.ibm.oti.util.Msg.getString("K0341")); //$NON-NLS-1$
-			}
-
-			/*[PR 115667, CMVC 94448] In 1.5, thread is added to ThreadGroup when started */
-			group.add(this);
-
+		synchronized (lock) {
```

Comment:
The lock cannot be dropped between testing `started` and the call to `startImpl()`, otherwise the same thread could be started multiple times.

Reply:
The field `started` is set by `startImpl()`, and both of them are within `synchronized void start()`, only one thread can enter, the same thread can't be started again.

Reply:
Ok, I forgot that the method is itself `synchronized`; my concern about deadlock is misplaced.